### PR TITLE
feat: Added custom slippage input for swapToken pair contract function

### PR DESCRIPTION
**Pull request by damiano-melcarne-lab49**
**Description**
- Added slippage parameter to the `swapToken` Pair contract function. This is intended to support the user's ability to customize slippage values at the transaction level in the UI.

### CODE REVIEW

File: contracts/Pair.sol [hunk 1]
```diff
@@ -9,6 +9,16 @@ import "./common/IERC20.sol";
 import "./ILPToken.sol";
 import "./IPair.sol";
 
+/// Emitted when the calculated slippage is over the slippage threshold.
+/// @param message a description of the error.
+/// @param calculatedSlippage the slippage calculated based on the requested transaction parameters.
+/// @param slippageThreshold the maximum slippage allowed for a transaction to proceed.
+error SlippageBreached(
```
Comment (damiano-melcarne-lab49) [inline contracts/Pair.sol:16]: More details on custom Solidity errors can be read here: https://blog.soliditylang.org/2021/04/21/custom-errors/

File: contracts/Pair.sol [hunk 2]
```diff
@@ -123,19 +134,19 @@ contract Pair is IPair, Initializable {
...
+        int256 _deltaAQty,
+        int256 _slippage
+    ) private {
         int256 calculatedSlippage = slippageOutGivenIn(_deltaAQty);
-        int256 localSlippage = getSlippage();
-        require(
-            calculatedSlippage <= (localSlippage),
-            "Slippage threshold breached."
-        );
+        isSlippageBreached(calculatedSlippage, _slippage);
```
Comment (amandeeprehal-lab49) [inline contracts/Pair.sol:149]: E2E tests are failing. - https://github.com/hashgraph/hedera-accelerator-defi-dex/actions/runs/4075156588/jobs/7021181481

Pls check.

File: contracts/Pair.sol (hunk 1 above)
Comment (amandeeprehal-lab49) [reply contracts/Pair.sol:16]: I like this, it will also help us to show these error on UI as its there in ABI.

File: contracts/Pair.sol [hunk 3]
```diff
@@ -493,7 +504,24 @@ contract Pair is IPair, Initializable {
         }
     }
 
-    function _tokenIsHBARX(address token) private returns(bool) {
+    function _tokenIsHBARX(address token) private returns (bool) {
         return token == tokenService.hbarxAddress();
     }
+
+    function isSlippageBreached(
```
Comment (amandeeprehal-lab49) [inline contracts/Pair.sol:511]: This method update state `slippageThreshold`.  `isSlippageBreached` name does not make sense.

File: contracts/Pair.sol [hunk 4]
```diff
@@ -493,7 +504,24 @@ contract Pair is IPair, Initializable {
...
+        int256 calculatedSlippage,
+        int256 _slippage
+    ) private view {
+        int256 slippageThreshold = getSlippage();
```
Comment (amandeeprehal-lab49) [inline contracts/Pair.sol:515]: You can write this as well -

`int256 slippageThreshold = _slippage > 0 ? _slippage:  getSlippage()`

File: contracts/Pair.sol (hunk 2 above)
Comment (damiano-melcarne-lab49) [reply contracts/Pair.sol:149]: I'm seeing multiple e2e test errors on **develop** as well. I'll run the test again after develop is stable.

**Review by amandeeprehal-lab49 — APPROVED** [review]

**Pull request merged**